### PR TITLE
cli_parse: Fix finding relative template_path when used inside role/collection

### LINKED ISSUE #332: cli_parse not finding relative template_path when used in role
##### SUMMARY

When `ansible.utils.cli_parse` is used in a role and `parser['temlate_path']` is defined as a relative path compared to the role directory (ex. inside `templates/` directory in the role) the module fails due to not finding the template file.
This type of usage is supported and commonly used in core Ansible modules like `template` or `copy` when they are used inside roles.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

cli_parse

##### ANSIBLE VERSION

```paste below
ansible [core 2.16.3]
  config file = None
  configured module search path = ['/home/vasim/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/vasim/ansible_testing/venv_ansible_latest/lib/python3.10/site-packages/ansible
  ansible collection location = /home/vasim/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/vasim/ansible_testing/venv_ansible_latest/bin/ansible
  python version = 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0] (/home/vasim/ansible_testing/venv_ansible_latest/bin/python3)
  jinja version = 3.1.3
  libyaml = True
```

##### COLLECTION VERSION

```paste below
Collection    Version
------------- -------
ansible.utils 2.12.0
```

##### STEPS TO REPRODUCE

```yaml
$ tree
.
├── roles
│   └── test_role
│       ├── tasks
│       │   └── main.yml
│       └── templates
│           └── test_data.yaml
└── test_playbook.yml

$ cat test_playbook.yml
---
- name: Use role with cli_parse template
  hosts: localhost
  gather_facts: false
  roles:
    - test_role

$ cat roles/test_role/tasks/main.yml
---
- name: use custom cli_parse template
  ansible.utils.cli_parse:
    text: |
      test data - foo
      test data - bar
    parser:
      name: ansible.netcommon.native
      template_path: "templates/test_data.yaml"

$ cat roles/test_role/templates/test_data.yaml
---
- example: test data - foo
  getval: 'test\sdata\s-\s(?P<data>.*)'
  result:
    data:
      - "{{ data }}"
```

##### EXPECTED RESULTS

cli_parse finds the template inside the role and parses the the given text

##### ACTUAL RESULTS

```paste below
$ ansible-playbook -v test_playbook.yml
No config file found; using defaults
[WARNING]: No inventory was parsed, only implicit localhost is available
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit
localhost does not match 'all'

PLAY [Use role with cli_parse template] **************************************************************

TASK [test_role : use custom cli_parse template] *****************************************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: FileNotFoundError: [Errno 2] No such file or directory: 'templates/test_data.yaml'
fatal: [localhost]: FAILED! => {"changed": false, "msg": "Failed to open template 'templates/test_data.yaml'. Error: [Errno 2] No such file or directory: 'templates/test_data.yaml'"}

PLAY RECAP *******************************************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0
```

If `template_path: "{{ role_path }}/templates/test_data.yaml"` is used instead then the template is found as expected, but this should not be necessary for cli_parse to behave consistently with core modules.

### PR DESCRIPTION
**Pull request by Mark0016**
##### SUMMARY

This allows the `cli_parse` module to find `parser.template_path` inside roles or collections when a path relative to the role/collection directory is provided.
This makes behavior consistent with core modules like `copy` and `template` when paths are provided this way.

Fixes #332

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

cli_parse

##### ADDITIONAL INFORMATION

See #332 for reproduction of the issue.
I used the same setup to validate that the changes fixed it.
With these changes cli_parse behaves as expected

```paste below
$ ansible-playbook -v test_playbook.yml
No config file found; using defaults
[WARNING]: No inventory was parsed, only implicit localhost is available
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit
localhost does not match 'all'

PLAY [Use role with cli_parse template] **************************************************************

TASK [test_role : use custom cli_parse template] *****************************************************
ok: [localhost] => {"changed": false, "parsed": {"data": ["foo", "bar"]}}

PLAY RECAP *******************************************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
```

### CODE REVIEW

File: tests/integration/inventory [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[testgroup]
```
Comment (NilashishC) [inline tests/integration/inventory:1]: Please remove this file.

File: tests/integration/targets/utils_cli_parse/output/ios_show_version_parsed.json [hunk 2]
```diff
@@ -0,0 +1,58 @@
+{
```
Comment (NilashishC) [inline tests/integration/targets/utils_cli_parse/output/ios_show_version_parsed.json:1]: I'm not sure I understand how the below parser would result in this data.

File: changelogs/fragments/add_template_path.yaml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+---
+minor_changes:
+  - Allows the cli_parse module to find parser.template_path inside roles or collections when a path relative to the role/collection directory is provided.
```
Comment (NilashishC) [inline changelogs/fragments/add_template_path.yaml:3]: Collections do not support a top-level "templates" directory, I believe. Are you sure this works?

**Review by NilashishC — CHANGES_REQUESTED** [review]

File: tests/integration/targets/utils_cli_parse/output/ios_show_version_parsed.json (hunk 2 above)
Comment (Ruchip16) [reply tests/integration/targets/utils_cli_parse/output/ios_show_version_parsed.json:1]: it parses the version in online and offline mode when we execute `show version` command
image

File: tests/integration/targets/utils_cli_parse/tasks/main.yaml [hunk 4]
```diff
@@ -8,6 +8,13 @@
   vars:
     parser: "({{ inventory_hostname }}/argspec)"
 
+- name: Include cli_parse.yaml
```
Comment (NilashishC) [inline tests/integration/targets/utils_cli_parse/tasks/main.yaml:11]: Minor comment: Can we improve the naming of this task and be more specific to _what_ we're testing here?

File: .github/workflows/tests.yml [hunk 5]
```diff
@@ -24,6 +30,11 @@ jobs:
     uses: ansible/ansible-content-actions/.github/workflows/unit.yaml@main
   integration:
     uses: ansible/ansible-content-actions/.github/workflows/integration.yaml@main
+  unit-source:
+    uses: ansible-network/github_actions/.github/workflows/unit_source.yml@main
```
Comment (NilashishC) [inline .github/workflows/tests.yml:34]:
```suggestion
    uses: Ruchip16/github_actions/.github/workflows/unit_source.yml@utils
```

File: .github/workflows/tests.yml [hunk 6]
```diff
@@ -11,6 +11,12 @@ on:  # yamllint disable-line rule:truthy
   workflow_dispatch:
   schedule:
     - cron: '0 0 * * *'
+  workflow_call:
```
Comment (NilashishC) [inline .github/workflows/tests.yml:14]: This isn't needed here.

**Review by Ruchip16 — APPROVED** [review]

**Review by NilashishC — APPROVED** [review]

**Pull request merged**